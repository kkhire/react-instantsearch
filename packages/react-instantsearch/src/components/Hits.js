import React, { createElement } from 'react';
import PropTypes from 'prop-types';

const Hits = ({ cx, hits, hitComponent, renderHit }) => (
  // Spread the hit on HitComponent instead of passing the full object. BC.
  // ex: <HitComponent {...hit} key={hit.objectID} />

  <ul className={cx('list')}>
    {hits.map(hit => (
      <li className={cx('item')} key={hit.objectID}>
        {renderHit ? renderHit(hit) : createElement(hitComponent, { hit })}
      </li>
    ))}
  </ul>
);

Hits.propTypes = {
  cx: PropTypes.func.isRequired,
  hits: PropTypes.array,
  hitComponent: PropTypes.func,
  renderHit: PropTypes.func,
};

Hits.defaultProps = {
  hitComponent: props => (
    <div
      style={{
        borderBottom: '1px solid #bbb',
        paddingBottom: '5px',
        marginBottom: '5px',
      }}
    >
      {JSON.stringify(props).slice(0, 100)}...
    </div>
  ),
};

export default Hits;
