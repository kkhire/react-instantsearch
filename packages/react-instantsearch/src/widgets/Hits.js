import React from 'react';
import PropTypes from 'prop-types';
import connectHits from '../connectors/connectHits';
import classNames from '../components/classNames';
import Panel from '../components/Panel';
import HitsComponent from '../components/Hits';

const cx = classNames('Hits');

/**
 * Displays a list of hits.
 *
 * To configure the number of hits being shown, use the [HitsPerPage widget](widgets/HitsPerPage.html),
 * [connectHitsPerPage connector](connectors/connectHitsPerPage.html) or the [Configure widget](widgets/Configure.html).
 *
 * @name Hits
 * @kind widget
 * @propType {function} [renderHit] - Function used for rendering each hit from
 *   the results. If it is not provided the rendering defaults to displaying the
 *   hit in its JSON form. The function will be called with `hit` as argument.
 * @propType {function} [renderHeader] - Adds a header to the widget.
 * @propType {function} [renderFooter] - Adds a footer to the widget.
 * @themeKey ais-Hits - the root div of the widget
 * @themeKey ais-Hits-header - the header of the widget (optional)
 * @themeKey ais-Hits-body - the body of the widget
 * @themeKey ais-Hits-list - the list of results
 * @themeKey ais-Hits-item - the hit list item
 * @themeKey ais-Hits-footer - the footer of the widget (optional)
 * @example
 * import React from 'react';

 * import { Hits, InstantSearch } from 'react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <Hits />
 *     </InstantSearch>
 *   );
 * }
 */

const Hits = ({ renderHeader, renderFooter, ...props }) => (
  <Panel cx={cx} renderHeader={renderHeader} renderFooter={renderFooter}>
    <HitsComponent {...props} cx={cx} />
  </Panel>
);

Hits.propTypes = {
  renderHeader: PropTypes.func,
  renderFooter: PropTypes.func,
};

export default connectHits(Hits);
