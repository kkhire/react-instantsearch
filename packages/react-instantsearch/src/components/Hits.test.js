import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Hits from './Hits';

const Hit = ({ hit }) => <div id={hit.objectID} />;

Hit.propTypes = {
  hit: PropTypes.object,
};

describe('Hits', () => {
  it('accepts a hitComponent prop', () => {
    const hits = [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }];

    const tree = renderer.create(
      <Hits cx={(...x) => x.join(' ')} hits={hits} hitComponent={Hit} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('accepts a renderHit prop', () => {
    const hits = [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }];

    const tree = renderer.create(
      <Hits
        cx={(...x) => x.join(' ')}
        hits={hits}
        renderHit={hit => <Hit hit={hit} />}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
