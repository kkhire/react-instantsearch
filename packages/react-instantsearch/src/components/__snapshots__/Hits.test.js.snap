// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits accepts a hitComponent prop 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <div
      id={0}
    />
  </li>
  <li
    className="item"
  >
    <div
      id={1}
    />
  </li>
  <li
    className="item"
  >
    <div
      id={2}
    />
  </li>
</ul>
`;

exports[`Hits accepts a renderHit prop 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <div
      id={0}
    />
  </li>
  <li
    className="item"
  >
    <div
      id={1}
    />
  </li>
  <li
    className="item"
  >
    <div
      id={2}
    />
  </li>
</ul>
`;
